services/proxy/pkg/middleware/oidc_auth.go

```diff
@@ -120,51 +104,25 @@ func (m *OIDCAuthenticator) getClaims(token string, req *http.Request) (map[stri
 			if err != nil {
 				m.Logger.Error().Err(err).Msg("failed to write to userinfo cache")
 			}
+
+			if sid, ok := claims["sid"]; ok {
+				// reuse user cache for session id lookup
+				err = m.userInfoCache.Write(&store.Record{
+					Key:    fmt.Sprintf("%s", sid),
+					Value:  []byte(encodedHash),
+					Expiry: time.Until(expiration),
+				})
+			}
+			if err != nil {
+				m.Logger.Error().Err(err).Msg("failed to write session lookup cache")
+			}
```

Comment:
This `err` check should be inside the `if`? Otherwise the `err` is checked twice in `!ok` case